entrypoint/remote/archive.go

```diff
@@ -0,0 +1,155 @@
+package remote
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"os"
+)
+
+func Archive(filePaths, directories []string, destination string) error {
```

Comment:
I suggest instead of taking args, there should be a `struct` and either take the struct or bind to it to help with readability. 

For example, 
```
type CosaArchive struct {
    CreateDirs []string
    IncludeDirs []string
    IncludePaths []string
}
```

And
```
func (a *CosaArchive) WriteArchive(dest string) error {....
```
Or
```
func CreateArchive(archive CosaArchive, dest string) error {....
```